docs/proposals/csi.md

```diff
@@ -0,0 +1,349 @@
+        - name:  csi-nfs-driver
+          securityContext:
+            privileged: true
+            capabilities:
+              add: ["SYS_ADMIN"]
+            allowPrivilegeEscalation: true
+          image: quay.io/k8scsi/csi-nfs-driver:v1.0.0
+          args :
+            - "--nodeid=$(NODE_ID)"
+            - "--endpoint=$(CSI_ENDPOINT)"
+          env:
+            - name: NODE_ID
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: CSI_ENDPOINT
+              value: unix://plugin/csi.sock
+          imagePullPolicy: "IfNotPresent"
+          volumeMounts:
+            - name: plugin-dir
+              mountPath: /plugin
+            - name: pods-mount-dir
+              mountPath: /var/lib/edged/pods
+              mountPropagation: "Bidirectional"
+      volumes:
+        - name: registration-dir
+          hostPath:
+            path: /var/lib/edged/plugins_registry
+            type: Directory
+        - name: plugin-dir
+          hostPath:
+            path: /var/lib/edged/plugins/csi-nfs-driver
+            type: DirectoryOrCreate
+        - name: pods-mount-dir
+          hostPath:
+            path: /var/lib/edged/pods
+            type: Directory
+```
+The `DaemonSet` includes:
+* The following containers:
+  * `node-driver-registrar`: responsible for registering the unix domain socket with edged.
+  * `csi-nfs-driver`: developped by the vendor.
+* The following volumes:
+  * `registration-dir` hostPath volume
+      Expose `/var/lib/edged/plugins_registry` from the edge node.
+      Mount only in `node-driver-registrar` container at `/registration`.
+      `node-driver-registrar` will use this unix domain socket to register the CSI driver’s unix domain socket with Edged.
+  * `plugin-dir` hostPath volume
+      Expose `/var/lib/edged/plugins/csi-nfs-driver/` from the edge node as `hostPath.type: DirectoryOrCreate`.
+      Mount inside `csi-nfs-driver` container which is the primary means of communication between Edged and the `csi-nfs-driver` container.
+  * `pods-mount-dir` hostPath volume
+      Expose `/var/lib/edged/pods` from the edge node.
+      Mount only in 'csi-nfs-driver' container at `/var/lib/edged/pods/`.
+      Ensure [bi-directional mount propagation](https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation) is enabled, 
+      so that any mounts setup inside this container are propagated back to edge node host machine.
+
+If it is not possible to deploy mutiple containers in  `DaemonSet` in some KubeEdge release versions,
+The cluster admins need to deploy two DaemonSets which includes `csi-nfs-driver` and `node-driver-registrar` respectively.
+
+### Example
+
+#### Static Provision Example
+
+Static Provision means the volume has already been created in somewhere.
+There is no need to create volume by the `External-Provisioner`.
+The customers can define it by PersistentVolume and PersistentVolumeClaim.
```

Comment:
Customers can request for persistent storage using PersistentVolume and PersistentVolumeClaim.

Reply:
Done